RecoTauTag/HLTProducers/src/L2TauTagNNProducer.cc

```diff
@@ -0,0 +1,891 @@
+         ebHandle = ec_tmp;
+       }
+       else if ((ec_tmp->begin()->detid()).subdetId() == EcalEndcap) {
+         eeHandle = ec_tmp;
+       }
+     }
+
+     std::vector<edm::EDGetTokenT<EcalRecHitCollection> >::const_iterator i;
+     for (i = ecal_tokens.begin(); i != ecal_tokens.end(); i++) {
+       edm::Handle<EcalRecHitCollection> ec;
+       event.getByToken(*i, ec);
+     }
+     edm::Handle<HBHERecHitCollection> hbhe;
+     event.getByToken(hbhe_token, hbhe);
+     edm::Handle<HORecHitCollection> ho;
+     event.getByToken(ho_token, ho);
+     edm::Handle<PixelTrackHeterogeneous> pataTracks;
+     event.getByToken(pataTracks_token, pataTracks);
+     edm::Handle<ZVertexHeterogeneous> pataVertices;
+     event.getByToken(pataVertices_token, pataVertices);
+     edm::Handle<reco::BeamSpot> bsHandle;
+     event.getByToken(BeamSpot_token, bsHandle);
+     edm::ESHandle<MagneticField> fieldESH;
+     eventsetup.get<IdealMagneticFieldRecord>().get(fieldESH);
+     edm::ESHandle<CaloGeometry> Geometry = eventsetup.getHandle(Geometry_token);
+     caloRecHitCollections caloRecHits;
+     caloRecHits.hbhe= &*hbhe;
+     caloRecHits.ho= &*ho;
+     caloRecHits.eb= &*ebHandle;
+     caloRecHits.ee= &*eeHandle;
+     caloRecHits.Geometry = &*Geometry;
+     int evt_id = event.id().event();
+
+     int nTaus = allTaus.size();
+     tensorflow::Tensor cellGridMatrix(tensorflow::DT_FLOAT, { nTaus, nCellEta, nCellPhi, nVars });
+     for (int tau_idx = 0; tau_idx<nTaus ; tau_idx++){
+       for (int eta_idx = 0; eta_idx<nCellEta ; eta_idx++){
+         for (int phi_idx = 0; phi_idx<nCellPhi ; phi_idx++){
+           for (int var_idx = 0; var_idx<nVars ; var_idx++){
+             cellGridMatrix.tensor<float, 4>()(tau_idx,phi_idx, eta_idx,var_idx) = 0.;
+           }
+         }
+       }
+     }
+     FillL1TauVars(cellGridMatrix, allTaus);
+
+     FillCaloRecHits(cellGridMatrix, allTaus, caloRecHits);
+
+     FillPatatracks(cellGridMatrix, allTaus, *pataTracks,*pataVertices,*bsHandle, fieldESH.product());
+
+     bool printoutevt=false;
+     if(debugLevel_>4){
+       if(evt_id==135505){
+         printoutevt=true;
+       }
+     }
+     standardizeTensor(cellGridMatrix); // do not remove this when eliminating debugging lines!!
+     checknan(cellGridMatrix,printoutevt, debugLevel_);
+
+     std::vector<float> tau_score = GetTauScore(cellGridMatrix);
+
+     for(size_t inp_idx = 0; inp_idx<L1TauDesc.size(); inp_idx++){
+       std::vector<float> outcomes_;
+       for(int tau_idx = 0; tau_idx < nTaus; tau_idx++){
+         std::cout <<  evt_id << " \t " << (allTaus[tau_idx])->polarP4().pt() << " \t "<< tau_score.at(tau_idx) << std::endl;
+         //std::cout << "evt " << evt_id << " tau Pt " << (allTaus[tau_idx])->polarP4().pt() << " score "<< tau_score.at(tau_idx) << std::endl;
+         if(std::find( TauCollectionMap.at(inp_idx).begin(), TauCollectionMap.at(inp_idx).end() ,allTaus[tau_idx]) != TauCollectionMap.at(inp_idx).end()){
```

Comment:
don't use `find` the second time. It is enough to use find once when you create allTaus at the beginning of the method.